build_tools/benchmarks/run_benchmarks_on_android.py

```diff
@@ -172,16 +175,17 @@ def adb_start_cmd(cmd_args: Sequence[str],
   return subprocess.Popen(cmd, stdout=subprocess.PIPE, text=True)
 
 
-def get_vmfb_full_path_for_benchmark_case(benchmark_case_dir: str) -> str:
-  flagfile_path = os.path.join(benchmark_case_dir, MODEL_FLAGFILE_NAME)
-  flagfile = open(flagfile_path, "r")
-  flagfile_lines = flagfile.readlines()
+def get_vmfb_full_path_for_benchmark_case(
+    benchmark_case_dir: pathlib.Path) -> pathlib.Path:
+  flagfile_path = benchmark_case_dir / MODEL_FLAGFILE_NAME
+  with flagfile_path.open("r") as flagfile:
+    flagfile_lines = flagfile.readlines()
   for line in flagfile_lines:
```

Comment:
Since you're reading all the lines at once anyway, maybe this is simpler?

```suggestion
  for line in flagfile_path.read_text().splitlines():
```

(note that this also drops the trailing newline).

Alternatively, if you don't want to bring the whole file into memory at once, you could bring the for loop into the with clause and just iterate over `flagfile`